regolith/helpers/l_contactshelper.py

```diff
@@ -0,0 +1,183 @@
+from regolith.helpers.basehelper import SoutHelperBase
+from regolith.fsclient import _id_key
+from regolith.tools import (
+    all_docs_from_collection,
+    get_pi_id,
+)
+
+TARGET_COLL = "contacts"
+HELPER_TARGET = "l_contacts"
+
+THIS_DIR = Path('.')
+inst_file = THIS_DIR / ".." / ".." / ".." / ".." / "dbs" / "rg-db-public" / "db" / "institutions.yml"
+
+def subparser(subpi):
+    subpi.add_argument("-n", "--name", help='match contacts to a name fragment')
+    subpi.add_argument("-i", "--inst", help='match people to institutions')
+    subpi.add_argument("-d", "--day", help='match people to days')
+    subpi.add_argument("-m", "--month", help='match people to months')
+    subpi.add_argument("-y", "--year", help='match people to years')
+    subpi.add_argument("-r", "--misc", help='match people to miscellaneous notes')
+    '''
+    subpi.add_argument("-v", "--verbose", action="store_true", help='increase verbosity of output')
+    subpi.add_argument("-l", "--lead",
+                       help="Filter milestones for this project lead"
+                       )
+    subpi.add_argument("-p", "--person",
+                       help="Filter milestones for this person whether lead or not"
+                       )
+    subpi.add_argument("-s", "--stati", nargs="+",
+                       help=f"List of stati for the project that you want returned,"
+                            f"from {ALLOWED_STATI}.  Default is proposed and started"
+                       )'''
+    return subpi
+
+
+class ContactsListerHelper(SoutHelperBase):
+    """Helper for listing upcoming (and past) projectum milestones.
+
+       Projecta are small bite-sized project quanta that typically will result in
+       one manuscript.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = HELPER_TARGET
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        try:
+            if not rc.database:
+                rc.database = rc.databases[0]["name"]
+        except:
+            pass
+        colls = [
+            sorted(
+                all_docs_from_collection(rc.client, collname), key=_id_key
+            )
+            for collname in self.needed_dbs
+        ]
+        for db, coll in zip(self.needed_dbs, colls):
+            gtx[db] = coll
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+
+    def sout(self):
+        rc = self.rc
+        contacts = []
+        '''
+        if rc.lead and rc.person:
+            raise RuntimeError(f"please specify either lead or person, not both")'''
+        for contact in self.gtx["contacts"]:
+            
```

Comment:
delete blank line (check PEP8 for syntax standards.  If you are using PyCharm it will format the file as best it can following pep8 rules.  Also you can install from conda-forge and run flake8 and/or black to lint the code automatically.